### ISSUE TITLE
[Web] ORT Training Doesn't Work for ResNet (Uncaught 51885088)

### ISSUE THREAD

**Issue by jaymody** [microsoft/onnxruntime#19811]
### Describe the issue

Attempting to load a ResNet model via `ort.TrainingSession` using `onnxruntime-web/training` leads to the following cryptic error:

```
Uncaught 51885088
```

### To reproduce

Code to reproduce can be found here: https://github.com/jaymody/ort-web-training-experiments/tree/resnet

```bash
git checkout -b resnet https://github.com/jaymody/ort-web-training-experiments
cd ort-web-training-experiments
python -m pip install torch torchvision onnx onnxruntime-training
python create_onnx_files.py "resnet"
npm install
npm run dev
```

I've also added the option to export a simple 2-layer network via `python create_onnx_files.py "mlp"` and that works fine, so at least to my knowledge, seems to be an architecture thing?

The resnet model is loaded using `torch.hub`.

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.17.1

### Execution Provider

'wasm'/'cpu' (WebAssembly CPU)

**Comment by carzh** [comment]
Hello, there are a handful of fixes:
* the vite.config.js src file path for loading models should be `models/**/*` to ensure that it will serve any files in nested directories under the models directory
* make sure that the paths in your js code align with the paths of the generated training artifacts
* Since the checkpoint implementation has changed in the previous versions of ORT, onnxruntime-web/training only works with training artifacts generated with the python package, onnxruntime-training 1.17.0+. The pip installation of onnxruntime-training gives you onnxruntime-training 1.15.0. Use the following installation command for latest stable onnxruntime-training package: `pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT/pypi/simple/ onnxruntime-training-cpu`. Other installation commands can be found [here](https://onnxruntime.ai/getting-started). 

After these steps, you will run into another error. There is a small bug with our training core code for resnet / batchnormalization. You can add the following code to the end of your create_onnx_files.py export script for the time being as a workaround: 
```
    eval_model = onnx.load("models/eval_model.onnx")

    for node in eval_model.graph.node:
        if node.op_type == "BatchNormalization":
            del node.output[1:]
    onnx.save(eval_model, "models/eval_model_updated.onnx")
```

After this, the training session should load properly in your website. 

If you run into issues during training or during later steps, you can try the following debug strategies:
* Write the training loop using the[ Python training APIs ](https://onnxruntime.ai/docs/api/python/on_device_training/training_api.html)to verify that the generated training artifacts work properly and that the data is in the correct shapes. 
* Use Netron to look at the generated ONNX model files and check that their inputs and outputs are the correct shapes
* If the training loop is working correctly with the python APIs, then the issue is in the web implementation. Make sure that the types of data and their shapes are correct. You can also build a debug version of onnxruntime-web from source.
